usr/src/cmd/ypcmd/ypxfr.c

```diff
@@ -357,8 +357,8 @@ logprintf(arg1, arg2, arg3, arg4, arg5, arg6, arg7)
 		(void) gettimeofday(&t, NULL);
 		(void) fprintf(stderr, "%19.19s: ", ctime(&t.tv_sec));
 	}
-	(void) fprintf(stderr, (char *)arg1, arg2, arg3, arg4, arg5,
-				arg6, arg7);
+	(void) fprintf(stderr, (char *)(uintptr_t)arg1, arg2, arg3,
+	    arg4, arg5, arg6, arg7);
```

Comment:
The underlying problem here is that the function arguments are not defined with a type, so have defaulted to `int` which is 32-bits for both 32- and 64-bit code. 64-bit has 64-bit pointers, and so the arguments are no longer big enough for the strings being passed in.
Casting through uintptr_t as you've done, will just put zeroes in the top 32-bits and does not actually fix things (apart from making the compiler quiet).
I would convert this to a varargs function.
```c
void
logprintf(const char *fmt, ...)
{
    va_list va;

    ....

    va_start(va, fmt);
    vfprintf(stderr, fmt, va);
    va_end(va);
}
```

Given this code is almost entirely K&R style, I would also skim through and check for any more places which have parameters without a declared type and fix them up.